expression/src/test/java/io/smallrye/common/expression/ExpressionTestCase.java

```diff
@@ -551,4 +551,22 @@ public void testPoint46() {
             assertEquals("", c.getExpandedDefault());
         }), "Should expand to empty string");
     }
+
+    @Test
+    void doubleColonFlagAndDefault() {
```

Comment:
Should be `testPoint7_5` or whatever to correlate above. Also missing are the following cases:

* `foo.bar::baz:something`
* `foo.bar::baz::something`
* `foo.bar:baz::something`
* `foo.bar:baz:something`

With, and without, `DOUBLE_COLON` set.

Reply:
I was wondering what should `foo.bar::baz:something` expand to. I think it makes sense that the resulting key is the entire expression since the `DOUBLE_COLON` flag states that the entire content is encoded into the key, but I can see people expecting to encode only `foo.bar::baz` in the key and use the default `something`.